### PR TITLE
WIP: Experiment with adding release jobs to image-registry

### PR DESCRIPTION
**Pull request by petr-muller**
This PR is an experiment about how we could add all release jobs to all OCP repositories, automatically. This PR tests the "melding" of ci-operator config for target repository (here, image registry) with the one where release jobs are defined. I only did this for one repository, and two (out of ~6) release job configs, but the principle should be the same. The melding needs to do the following:
- `base_images` is a union of the two configs
- `tests` is taken from the release job config - but the jobs must be turned into presubmits from periodics!
- `releases` are taken from the release job config - except for `latest` release which is filtered out
- `tag_specification` is taken from the component config
- `images` are taken from component config
- `build_root` is taken from component config
- `resources` should be probably a union of the two, with using higher values of the two for the cases where the two configs define a same key

Note that we cannot currently automatically mark the jobs as non-blocking/triggered manually - we need [DPTP-2301](https://issues.redhat.com/browse/DPTP-2301) / [DPTP-1973](https://issues.redhat.com/browse/DPTP-1973) / [DPTP-1364](https://issues.redhat.com/browse/DPTP-1364) for that. 

Question is, how to scale this to all repos / all branches / all release jobs, of course this cannot be maintained manually.
- We could write a tool that would maintain these variant ci-operator config files (ones under `ci-operator/config`): they would then be normally taken by prowgen, generated jobs, etc. This is probably not that hard to do, but it would result in a *lot* of clutter in the repo, alongside the files that we expect people to touch.
- We can fixup Prowgen to only generate the `ci-operator/jobs` files (actual per-repo job configuration), even when the source "melded" ci-operator configs (ci-operator/config) files do not exist. The problem here is that these jobs would have nothing to `--target`, they basically back-link to ci-operator config files. The solution here could be to do the "melding" on-demand, in ci-operator-configresolver. We can take advantage of the fact that ci-operator executions in CI jobs are getting their ci-operator configs from config-resolver (basically a config-serving server), so we can ask the resolver to give us a melded config that we can execute.

I think the latter is feasible. It still results in a TON of clutter in openshift/release, but under `ci-operator/jobs` - we don't want people to interact with these (and they will eventually go away in DPTP-1304)

### CODE REVIEW

**Comment by petr-muller** [comment]
Note: in the current revision, the jobs do not actually test the built version if image-registry, but they test directly the releases in `releases`. We cannot assemble an ephemeral release if we are not using `tag_specification`. We could use tag_specification, but that would end up testing something else than a nightly.

**Comment by petr-muller** [comment]
Upgrade tests are behaving weirdly, there are probably some bugs.

**Comment by deads2k** [comment]
Is there code somewhere doing the merging from your description?

**Comment by petr-muller** [comment]
@deads2k just local so far

**Comment by petr-muller** [comment]
Thread about using minor-to-current upgrade jobs on PRs: https://coreos.slack.com/archives/C01CQA76KMX/p1626954559451200

TL;DR there are bugs: 
- https://issues.redhat.com/browse/DPTP-2395
- https://issues.redhat.com/browse/DPTP-2396
